src/react/utils.ts

```diff
@@ -0,0 +1,11 @@
+export function shouldThrowError<TError>(
+  suspense: boolean | undefined,
+  _useErrorBoundary: boolean | ((err: TError) => boolean) | undefined,
+  error: TError
+): boolean {
+  if (typeof _useErrorBoundary === 'function') {
+    return _useErrorBoundary(error)
```

Comment:
just one more thought: would it also make sense to maybe pass `currentResult.data` here? One use-case would be to only use error boundaries if you get an error but don't have data yet, and you could decide to not throw if you already have data, because stale data might be better than unmounting everything and showing an error boundary instead. what do you think?

Reply:
I don't know enough about react-query's behaviour regarding the result.data state when the query is in an error state to say.

I'm not sure about stuffing the arguments of the function right now. Maybe we should wait till we know all the potential hypothetical use cases. If there are reasons a user might want other parts of the result besides error or data then we could also change the function to `(error: TError, result: QueryObserverResult<TData, TError>) => boolean` later without a breaking change.

Reply:
That sounds very reasonable 👍